### PR TITLE
Added Dependency Injection

### PR DESCRIPTION
**Pull request by vglalitha**

### CODE REVIEW

File: dropwizard-demo/src/main/java/com/dep/models/SendResponse.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    public int getStatusCode() {
+        return statusCode;
+    }
+
+    public void setStatusCode(int statusCode) {
+        this.statusCode = statusCode;
+    }
+
+    private int statusCode;
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/models/SendResponse.java]: Declared the variables together at class level

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 2]
```diff
@@ -1,80 +1,74 @@
...
-        this.twitterImplement = twitterImplement;
-    }
 
-    public Controller() {
-        twitterImplement = new TwitterImplement();
-    }
-
-    @GET
-    @Path("/getTweets")
+    @GetMapping("/getTweets")
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: Use either GetMapping or RequestMapping in ur project

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 3]
```diff
@@ -1,80 +1,74 @@
...
     public Response fetchTweets() {
         return Response.ok(twitterImplement.fetchLatestTweets()).build();
     }
 
-    @GET
-    @Path("/filteredTweets")
+
+    @RequestMapping("/filteredTweets/{searchKey}")
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: only path is mentioned, specifically mentioning HTTP method will make the API more robust

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 4]
```diff
@@ -1,80 +1,74 @@
...
-    }
 
-    public Controller() {
-        twitterImplement = new TwitterImplement();
-    }
-
-    @GET
-    @Path("/getTweets")
+    @GetMapping("/getTweets")
     public Response fetchTweets() {
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: fetch tweets still returning Response ? did u not changed as a part of https://confluence.dev.lithium.com/display/CD/Lab%3A+Pojos task ?

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 5]
```diff
@@ -1,80 +1,74 @@
...
     public String healthCheck() {
         return "Ping Received at " + new Date();
     }
 
-    @POST
-    @Path("/tweetAgain")
-    @Consumes(MediaType.APPLICATION_JSON)
-    public Response sendTweet(Request request) throws TwitterException {
+
+    @RequestMapping(method = RequestMethod.POST, value = "/tweetAgain")
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java:55]: Implement this in all of ur RequestMapping annotation

File: dropwizard-demo/test/com/test/TwitterImplementTest.java [hunk 6]
```diff
@@ -7,6 +7,7 @@
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
```
Comment (anasuya1117) [inline dropwizard-demo/test/com/test/TwitterImplementTest.java]: not used, remove it.

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java (hunk 2 above)
Comment (vglalitha) [reply dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: addressed review

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java (hunk 4 above)
Comment (vglalitha) [reply dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: changed  return format from Response to List.

File: dropwizard-demo/test/com/test/TwitterImplementTest.java (hunk 6 above)
Comment (vglalitha) [reply dropwizard-demo/test/com/test/TwitterImplementTest.java]: removed.

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 7]
```diff
@@ -26,52 +28,49 @@
     @Autowired
     TwitterImplement twitterImplement;
 
-
-    @GetMapping("/getTweets")
-    public Response fetchTweets() {
-        return Response.ok(twitterImplement.fetchLatestTweets()).build();
+    @RequestMapping("/getTweets")
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: how do i understand its a GET/POST/DELETE Api ?

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 8]
```diff
@@ -26,52 +28,49 @@
...
+                return new SendResponse("Please enter valid tweet", 400);
             } else {
-                logger.error("internal error occurred");
-                return new SendResponse("internal server error", 500);
+                Status status = twitterImplement.sendTweets(post);
+                if (status.getText().equals(post)) {
+                    logger.info("successfully posted");
+                    return new SendResponse("Tweet posted Successfully", 200);
+                } else {
+                    logger.error("internal error occurred");
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: how can this be a `internal server error`, program will come to this place only if tweet is not successful from `twitterImplement.sendTweets`, in that case its a problem of twitter library. Also `internal server error` message should be something understandable , its technical term, please update it.

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 9]
```diff
@@ -26,52 +28,49 @@
...
+                    return new SendResponse("internal server error", 500);
+                }
             }
+        } catch (Exception e) {
+            System.out.println(e);
+            return null;
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: returning null ??? have u tried this case ? a test case maybe ?

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java (hunk 3 above)
Comment (anasuya1117) [reply dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: This is still not addresses

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java (hunk 8 above)
Comment (vglalitha) [reply dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: Changed the message to "Request is incomplete"

File: dropwizard-demo/src/main/java/com/dep/models/SendResponse.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
+package com.dep.models;
+
+public class SendResponse {
+    private String message;
+    private int statusCode;
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/models/SendResponse.java]: not needed, can be removed

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 11]
```diff
@@ -1,83 +1,73 @@
...
-        return Response.ok(twitterImplement.fetchLatestTweets()).build();
+    @RequestMapping(method = RequestMethod.GET, value = "/getTweets")
+    public List<TweetResponse> fetchTweets() {
+        return twitterImplement.fetchLatestTweets();
     }
 
-    @GET
-    @Path("/filteredTweets")
+    @RequestMapping(method = RequestMethod.GET, value = "/filteredTweets")
     @Consumes(MediaType.APPLICATION_JSON)
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: if not needed, remove

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 12]
```diff
@@ -1,83 +1,73 @@
...
-        if (StringUtil.isEmpty(post)) {
-            logger.error("error happened");
-            return Response.status(400, "Please enter valid tweet").build();
-        } else {
-            Status status = twitterImplement.sendTweets(post);
-            if (status.getText().equals(post)) {
-                logger.info("successfully posted");
-                return Response.status(200, "Request is successful").build();
+    @RequestMapping(method = RequestMethod.POST, value = "/tweetAgain")
+    public SendResponse sendTweet(@RequestBody Request request) throws TwitterException {
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: Update it to ResponseEntity<SendResponse>

**Review by anasuya1117 — APPROVED** [review]
Address the comments and merge

**Pull request merged**